[#178] feat: 장바구니 진열장 업데이트 메소드 삽입

gustn3965 opened the pull request:
## 개요
-  장바구니 진열장 업데이트 메소드 삽입

![dra](https://user-images.githubusercontent.com/48749182/144032596-39f51d4b-e408-47d0-befe-1eca952cba59.gif)


## 작업 사항
- 파일 하나만 수정했습니다
    -  ~~`PostViewController+DataSource.swift` 에서 샀어요 버튼을 누르면, `ViewModel`에 `pageType`을 `.bought`으로 전달하는 것을 보고,  `WriteViewController`에서 `Done` 버튼을 누를 경우,  `viewModel`을 체크하는 메소드를 추가하여,  장바구니 업데이트 메소드를 삽입했습니다.~~

## 추가작업 
-  기존에 구현한 체크하는 메소드 를 `WriteViewModel`에 옮김.
- `modifyEntity`를 그대로 이용하게 되면, `save` 이후에 `modifyEntity`의 타입이 변경되므로, 변경되지 않는 `modifyEntityPageType`의 프로퍼티를 `WriteViewModel`에 갖도록 했고, 이를 조건으로 판단함

### Linked Issue

close #178

gustn3965 committed:
[#178] feat: 장바구니 진열장 업데이트 메소드 삽입

jwonyLee reviewed (commented):
`WriteViewController`에서 단순히 `viewModel.pageType` 만 체크하게 되면 `샀다` 게시물을 작성한 경우에도 해당 조건을 만족하게 됩니다. 제가 생각하기에는 `checkIsFromBoughtButton()`를 `WriteViewModel`로 옮기고, `게시물을 수정할 때 && pageType 을 수정하기 전`에 해당 메서드를 호출하면 될 거 같습니당.

```swift
func updatePost() {
  // 생략
  checkIsFromBoughtButton()
  modifyEntity.postType?.pageType = pageType
  // 생략
}
```

```swift
private func checkIsFromBoughtButton() {
    if modifyEntity?.postType?.pageType == .wish && pageType == .bought {
        DrawerCoreDataRepository(coreDataStack: CoreDataStack.shared).updateBasket()
    }
}
```

gustn3965 committed:
[#178] ref: 지원님 피드백 반영
- 기존에 구현한 체크하는 메소드를 WriteViewModel에 옮김.
- modifyEntity를 그대로 이용하게 되면, save 이후에 modifyEntity의 타입이 변경되므로, 변경되지 않는 modifyEntityPageType의 프로퍼티를 WriteViewModel에 갖도록 했고, 이를 조건으로 판단함

gustn3965 commented:
지원님 리뷰를 반영했습니다.  지원님이 제안하신 코드를 그대로 적용했을 때는, `save` 메소드 이후에 해당 코드를 실행시키면, `modifyEntity`가 변경된 상태라 기대하는 결과가 나타나지 않았습니다.  그래서 변경되지 않는 값으로 프로퍼티를 갖도록 했습니다.
- `modifyEntity`를 그대로 이용하게 되면, `save` 이후에 `modifyEntity`의 타입이 변경되므로, 변경되지 않는 `modifyEntityPageType`의 프로퍼티를 `WriteViewModel`에 갖도록 했고, 이를 조건으로 판단함

jwonyLee reviewed (approved):
LGTM 👍
